File: day11_part1_2017.py
with open("input.txt", "r") as file:
    directions = file.read().strip().split(",")

x = y = z = 0
max_distance = 0

for direction in directions:
    if direction == "n":
        y += 1
        z -= 1
    elif direction == "ne":
        x += 1
        z -= 1
    elif direction == "se":
        x += 1
        y -= 1
    elif direction == "s":
        y -= 1
        z += 1
    elif direction == "sw":
        x -= 1
        z += 1
    elif direction == "nw":
        x -= 1
        y += 1
    
    max_distance = max(max(abs(x), abs(y), abs(z)), max_distance)

result = max(abs(x), abs(y), abs(z))
print(result)

File: training_data/reflection-tuning/day11_part1_2017.go
package main

import (
	"fmt"
	"io/ioutil"
	"math"
	"strings"
)

func main() {
	// Read input
	input, err := ioutil.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}

	// Parse steps
	steps := strings.Split(strings.TrimSpace(string(input)), ",")

	// Initialize position
	x, y, z := 0, 0, 0

	// Process steps
	for _, step := range steps {
		switch step {
		case "n":
			y++
			z--
		case "ne":
			x++
			z--
		case "se":
			x++
			y--
		case "s":
			y--
			z++
		case "sw":
			x--
			z++
		case "nw":
			x--
			y++
		}
	}

	// Calculate shortest path
	distance := math.Max(math.Abs(float64(x)), math.Max(math.Abs(float64(y)), math.Abs(float64(z))))

	fmt.Printf("Fewest steps required: %.0f\n", distance)
}
